File: src/biaobiaoqi/algorithm/oj/pat/advancedlevel/APAT1040.cpp
#include <stdio.h>
#include <stdlib.h>
#define max 100005

//http://pat.zju.edu.cn/contests/pat-a-practise/1013

int main() {
	int n, m;
	int tmp;
	int start, end;
	int distance;
	int dis[max];
	scanf("%d",	&n);
	dis[0] = 0;
	for (int i = 1; i <= n; i ++){
		scanf("%d", &tmp);
		dis[i] = dis[i - 1] + tmp;
	}
	scanf("%d", &m);
	for (int i = 0; i != m; i ++){
		scanf("%d%d", &start, &end);
		if (start > end){
			tmp = end;
			end = start;
			start = tmp;
		}
		distance = dis[end - 1] - dis[start - 1];
		if (distance > dis[n] - distance)
			distance = dis[n] - distance;

		printf("%d\n", distance);
	}
	return 0;
}

File: src/biaobiaoqi/algorithm/oj/pat/advancedlevel/APAT1040.java
package biaobiaoqi.algorithm.oj.pat.advancedlevel;

import java.util.Scanner;
//http://pat.zju.edu.cn/contests/pat-a-practise/1040
//time out. cpp format can be accepted.

public class APAT1040 {
 
	
	public static void main(String[] args){
		Scanner cin = new Scanner(System.in);
		String str = cin.nextLine();
		int max = 0;
		int length = str.length();
		for (int i = 0; i != str.length(); i ++){
			//'aba', take care the lower limit is 0.
			int j = 0;
			while(i-j >= 0 && i+j < length && str.charAt(i-j) == str.charAt(i+j)){
				if (max < j*2 + 1)
					max = j*2 + 1;
				j ++;
			}
			
			//i as the former 'b' in 'abba'
			j = 1;
			while(i + 1- j >= 0 && i + j < length && str.charAt(i + 1 - j) == str.charAt(i + j)){
				if (max < j*2)
					max = j*2;
				j ++;
			}
		}
		
		System.out.println(max);
	}

	 
}